examples/grovers.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iomanip> // For setw
+#include <iostream> // For cout
+
+// "qfactory.hpp" pulls in all headers needed to create any type of "Qrack::QInterface."
+#include "qfactory.hpp"
+
+using namespace Qrack;
+
+int main()
+{
+// ***Grover's search, to invert a black box function***
+
+// Both CPU and GPU types share the QInterface API.
+#if ENABLE_OPENCL
+    QInterfacePtr qReg = CreateQuantumInterface(QINTERFACE_OPENCL, 20, 0);
+#else
+    QInterfacePtr qReg = CreateQuantumInterface(QINTERFACE_CPU, 20, 0);
+#endif
+
+    int i;
+
+    // Our subroutine returns true only for an input of 100.
+
+    const int TARGET_PROB = 100;
```

Comment:
Should be a parameter to a function, with this separated from the contents of main.

Reply:
However, this does make sense for the lookup table variant, and I've done so there.